### PR TITLE
Components: ClipboardButton

### PR DESCRIPTION
**Pull request by saramarcondes**
Add a new clipboard button that handles copying and a hook for when copying is finished.

No screenshots as it's just a button with some special props.

### CODE REVIEW

File: packages/components/src/ClipboardButton/ClipboardButton.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		if (hasCopied) {
+			onCopyProp();
+		} else {
+			onFinishCopy();
+		}
+
+		lastHasCopied.current = hasCopied;
+	}, [hasCopied, onCopyProp, onFinishCopy]);
+
+	return <Button {...otherProps} onClick={onCopy} ref={forwardedRef} />;
```
Comment (ItsJonQ) [inline packages/components/src/ClipboardButton/ClipboardButton.js:34]: @saramarcondes Would love your thoughts on this. I think we should pass-through an `onClick` callback as well (from props), in addition to firing the `onCopy` callback. I'd expect `onClick` to work (as a consumer), even though under the hood it's firing `onCopy`. What do you think :)

Comment (saramarcondes) [reply packages/components/src/ClipboardButton/ClipboardButton.js:34]: I had the same thought, I'm glad you brought it up. I think it makes sense to merge the two as you suggested 👍

File: packages/components/src/ClipboardButton/ClipboardButton.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import React from 'react';
+import { useEffect } from 'react';
+import { useRef } from 'react';
+
+import { Button } from '../Button';
+
+function ClipboardButton(props, forwardedRef) {
+	const {
+		text,
+		onClick: onClickProp = noop,
```
Comment (ItsJonQ) [inline packages/components/src/ClipboardButton/ClipboardButton.js:13]: I don't think we need to alias `onClick` to `onClickProp` here :)
(unlike `onCopy`, which comes from `useClipboard`)

File: packages/components/src/ClipboardButton/ClipboardButton.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		if (hasCopied) {
+			onCopyProp();
+		} else {
+			onFinishCopy();
+		}
+
+		lastHasCopied.current = hasCopied;
+	}, [hasCopied, onCopyProp, onFinishCopy]);
+
+	const handleOnClick = (event) => {
```
Comment (ItsJonQ) [inline packages/components/src/ClipboardButton/ClipboardButton.js:35]: @saramarcondes I've been memoizing callbacks G2 Components. I didn't originally... how important it was for [performance](https://github.com/ItsJonQ/g2/issues/57) 🙈 (assuming the consumer also passes in a memoized callback).

It's slightly tedious, but I think the trade off and value is worth it 🤞

**Comment by ItsJonQ** [comment]
@saramarcondes Haii! I left some comments :).

The last thing we gotta do is export this component in the main `src/index.js`.

Unfortunately, the `remake` command doesn't automatically add it in the `src/index.js` file 😊

**Review by ItsJonQ — APPROVED** [review]
🚀 from me! Thanks @saramarcondes !!!

**Pull request merged**